packages/flutter_tools/lib/src/windows/visual_studio_project.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:xml/xml.dart' as xml;
+
+import '../base/file_system.dart';
+import '../globals.dart' as globals;
+
+/// A utility class for interacting with Visual Studio project files (e.g.,
+/// .vcxproj).
+class VisualStudioProject {
```

Comment:
Rather than using the global file system, I would inject a fileSystem instance in the constructor. We've been trying to update many of the tool interfaces to work this way, because it makes it easier to tell what these classes depend on for testing

Reply:
Done. Should I do the same thing for the stand-alone solution file utility function (as a parameter of the function)? It's a big ugly as I'd need to plumb it through some helper calls, unless I wrap the whole thing in a class (which I can certainly do if that's the preferred structure going forward).

Reply:
Sorry, didn't see your later messages until now. Will rework this now.